### PR TITLE
Add an integration test

### PR DESCRIPTION
**Pull request by hundt-corbalt**
Includes some slight refactoring to make testing easier

### CODE REVIEW

File: main_test.go [hunk 1]
```diff
@@ -0,0 +1,714 @@
...
+					t.Fatalf("%s row %d: user %q is not in password manager",
+						sheetNameMACFin, idx+1, username)
+				} else if expectedPassword != password {
+					t.Fatalf("%s row %d: Expected password %q but got %q",
+						sheetNameMACFin, idx+1, expectedPassword, password)
+				}
+				delete(macFinUsers, username)
+			}
+			if len(macFinUsers) != 0 {
+				t.Fatalf("Users are now missing from the MACFin sheet: %v", macFinUsers)
```
Comment (leslie-corbalt) [inline main_test.go:691]: Should this be "Users are now missing from the password manager sheet?"

**Review by leslie-corbalt — COMMENTED** [review]
Looks great! Just a few questions.

File: main_test.go [hunk 2]
```diff
@@ -0,0 +1,714 @@
...
+	xsrfToken string
+	loggedOut bool
+}
+
+type AuthServer struct {
+	UserToPassword    map[string]string
+	UserToNewPassword map[string]string
+	Errors            map[string]string // user -> path
+	PortalSessions    map[string]*session
+	IMDSessions       map[string]*session
```
[the remark this answers is not in the document]
Comment (hundt-corbalt) [reply main_test.go:78]: Typo

File: main_test.go [hunk 3]
```diff
@@ -0,0 +1,714 @@
...
+
+func TestRotate(t *testing.T) {
+	for _, tc := range testCases {
+		t.Run(tc.Name, func(t *testing.T) {
+			dir, err := os.MkdirTemp(os.TempDir(), "test-user-manager")
+			if err != nil {
+				t.Fatalf("Error making temp dir: %s", err)
+			}
+			log.Printf("Created %s", dir)
+			// defer os.RemoveAll(dir)
```
[the remark this answers is not in the document]
Comment (hundt-corbalt) [reply main_test.go:521]: Oops, had it commented out for debugging, thanks

File: main_test.go (hunk 1 above)
Comment (hundt-corbalt) [reply main_test.go:691]: No, that would be found when we check `PasswordManagerOut`

**Review by leslie-corbalt — APPROVED** [review]
Looks great to me!

**Pull request merged**